app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepsummary/SummaryDestination.kt

```diff
@@ -1,18 +1,345 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.hedvig.android.data.changetier.data.TierDeductibleQuote
+import com.hedvig.android.design.system.hedvig.ButtonDefaults.ButtonSize
+import com.hedvig.android.design.system.hedvig.ButtonDefaults.ButtonSize.Large
+import com.hedvig.android.design.system.hedvig.ButtonDefaults.ButtonStyle.Primary
+import com.hedvig.android.design.system.hedvig.ButtonDefaults.ButtonStyle.Secondary
+import com.hedvig.android.design.system.hedvig.HedvigButton
+import com.hedvig.android.design.system.hedvig.HedvigErrorSection
+import com.hedvig.android.design.system.hedvig.HedvigFullScreenCenterAlignedLinearProgress
+import com.hedvig.android.design.system.hedvig.HedvigFullScreenCenterAlignedProgress
+import com.hedvig.android.design.system.hedvig.HedvigScaffold
 import com.hedvig.android.design.system.hedvig.HedvigText
+import com.hedvig.android.design.system.hedvig.HedvigTheme
+import com.hedvig.android.design.system.hedvig.HedvigThreeDotsProgressIndicator
+import com.hedvig.android.design.system.hedvig.HorizontalDivider
+import com.hedvig.android.design.system.hedvig.HorizontalItemsWithMaximumSpaceTaken
+import com.hedvig.android.design.system.hedvig.Icon
+import com.hedvig.android.design.system.hedvig.IconButton
+import com.hedvig.android.design.system.hedvig.Surface
+import com.hedvig.android.design.system.hedvig.icon.ArrowNorthEast
+import com.hedvig.android.design.system.hedvig.icon.HedvigIcons
+import com.hedvig.android.feature.change.tier.ui.stepcustomize.PillAndBasicInfo
+import com.hedvig.android.feature.change.tier.ui.stepsummary.SummaryState.Failure
+import com.hedvig.android.feature.change.tier.ui.stepsummary.SummaryState.Loading
+import com.hedvig.android.feature.change.tier.ui.stepsummary.SummaryState.MakingChanges
+import com.hedvig.android.feature.change.tier.ui.stepsummary.SummaryState.Success
+import hedvig.resources.R
+import java.io.File
 
 @Composable
 internal fun ChangeTierSummaryDestination(
   viewModel: SummaryViewModel,
   navigateUp: () -> Unit,
-  onNavigateToNewConversation: () -> Unit,
-  openUrl: (String) -> Unit,
+  onSuccess: () -> Unit,
+  onFailure: () -> Unit,
+  sharePdf: (File) -> Unit,
 ) {
-  Box(contentAlignment = Alignment.Center) {
-    HedvigText("Summary")
+  val uiState: SummaryState by viewModel.uiState.collectAsStateWithLifecycle()
+  when (uiState) {
```

Comment:
Can we just do the `collectAsStateWithLifecycle` here and then delegate to another `ChangeTierSummaryScreen` composable which we can then `@Preview` all together for all possible SummaryState states?